server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java

```diff
@@ -305,7 +305,12 @@ public void dotestSimpleTerminateAfterCountWithSize(int size, int max) throws Ex
                 .setSize(size)
                 .setTrackTotalHits(true)
                 .get();
-            assertHitCount(searchResponse, i);
+
+            if (size == 0) {
+                assertHitCount(searchResponse, i, max);
```

Comment:
@austintlee could you please add a reference to Apache Lucene pull request or ticket that explains the behaviour?  thank you

Reply:
Will do.